shared/zodSchemas.ts

```diff
@@ -71,6 +71,18 @@ export const createSessionSchema = z.object({
   user: z.custom<Prisma.UserCreateNestedOneWithoutSessionInput>(),
 });
 
+export const feedbackSchema = z.object({
+  feedbackUUID: z.string(),
+  feedbackText: z.string().min(1, 'Feedback text is mandatory!'),
+  createdAt: z.date(),
+  updatedAt: z.date(),
+});
+
+export const createFeedbackSchema = feedbackSchema.pick({
+  feedbackText: true,
+  feedbackUUID: true,
+});
```

Comment:
Tässä tämä `feedbackUUID`-nimitys ei ole mielestäni hyvä, koska kyseessä on `feedbackSessionUUID` (istunnon `UUID`), ei palautteen `UUID`.

Voisiko näin periaatteessa tehdä:

```suggestion
export const createFeedbackSchema = z.object({
  feedbackText: z.string().min(1, "Feedback text is mandatory!"),
  feedbackSessionUUID: z.string().min(1, "Feedback session UUID is mandatory!")
});
```

Eli ettei käyttäisikään `feedbackSchema`a pohjana, vaan tekisi täysin oman scheman käyttämällä `z.object()`:ia?

Ainakin omaan silmään tuo `feedbackSessionUUID` olisi parempi nimitys.


Sen verran vielä, että tämä tosiaan on tarpeellinen vain siinä kohtaa, jos `feedbackSession` aiotaan säilyttää :smile:

Reply:
Ja sen verran vielä, että ehdottamani tapa ei "seuraisi" muita `createNimiSchema`-tapoja.

Esimerkki `userSchema`:sta

```ts
export const userSchema = z.object({
  email: emailSchema,
  firstName: firstNameSchema,
  lastName: lastNameSchema,
});

export const createUserSchema = userSchema.extend({
  password: passwordSchema,
});
```
tässä `createUserSchema` käyttää `userSchema`a "pohjana"

Reply:
Otetaas vielä kerran. Tuossa kun kävin asioilla, tuli mieleen tämmöinen ratkaisutapa :smile: 

```ts
// Tässä variablessa on vain feedbackText, koska se on oleellisin tieto mitä palautteen pitäisi yleisesti sisältää. Muut tiedot eivät ole niin tärkeitä
export const feedbackSchema = z.object({
  feedbackText: z.string().min(1, 'Feedback text is mandatory!'),
});

// Tähän voidaan lisätä ID, UUID, createdAt ja updatedAt, koska tätä tapaa voi käyttää vain ylläpitäjät
export const getFeedbackSchema = feedbackSchema.extend({
  feedbackID: z.number().min(1, 'Feedback ID is mandatory!'),
  feedbackUUID: z.string().min(1, 'Feedback ID is mandatory!'),
  createdAt: z.date(),
  updatedAt: z.date(),
});

// Kun palautetta lähetetään, tarvitaan teksti (feedbackSchemasta feedbackText) ja feedbackSessionUUID, jolla varmistetaan voidaanko palaute ottaa vastaan
export const createFeedbackSchema = feedbackSchema.extend({
  feedbackSessionUUID: z
    .string()
    .min(1, "Feedback session's UUID is mandatory!"),
});
```


Tämä tosiaan voi olla todella sekava, mutta kysäse niin yritän selittää parhaani mukaan mitä hain takaa :smile: